x/accesscontrol/module.go

```diff
@@ -0,0 +1,169 @@
+
+// RegisterRESTRoutes registers the REST routes for the accesscontrol module.
+func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Router) {}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the accesscontrol module.
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {}
+
+// GetTxCmd returns no root tx command for the accesscontrol module.
+func (AppModuleBasic) GetTxCmd() *cobra.Command { return nil }
+
+// GetQueryCmd returns the root query command for the accesscontrol module.
+func (AppModuleBasic) GetQueryCmd() *cobra.Command { return nil }
+
+// AppModule implements an application module for the accesscontrol module.
+type AppModule struct {
+	AppModuleBasic
+
+	keeper keeper.Keeper
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(cdc codec.Codec, keeper keeper.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{cdc: cdc},
+		keeper:         keeper,
+	}
+}
+
+// Name returns the accesscontrol module's name.
+func (AppModule) Name() string {
+	return types.ModuleName
+}
+
+// RegisterInvariants registers the accesscontrol module invariants.
+func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+
+// Route returns the message routing key for the accesscontrol module.
+func (AppModule) Route() sdk.Route { return sdk.Route{} }
+
+// QuerierRoute returns the accesscontrol module's querier route name.
+func (AppModule) QuerierRoute() string {
+	return types.QuerierRoute
+}
+
+// LegacyQuerierHandler returns the x/accesscontrol module's sdk.Querier.
+func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
+	return func(sdk.Context, []string, abci.RequestQuery) ([]byte, error) {
+		return nil, fmt.Errorf("legacy querier not supported for the x/%s module", types.ModuleName)
+	}
+}
+
+// RegisterServices registers a gRPC query service to respond to the
+// module-specific gRPC queries.
+func (am AppModule) RegisterServices(cfg module.Configurator) {}
+
+// InitGenesis performs genesis initialization for the accesscontrol module. It returns
+// no validator updates.
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
+	var genState types.GenesisState
+	cdc.MustUnmarshalJSON(data, &genState)
+
+	am.keeper.InitGenesis(ctx, genState)
+
+	return []abci.ValidatorUpdate{}
+}
+
+// ExportGenesis returns the exported genesis state as raw bytes for the mint
+// module.
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
+	genState := am.keeper.ExportGenesis(ctx)
+	return cdc.MustMarshalJSON(genState)
+}
+
+// ConsensusVersion implements AppModule/ConsensusVersion.
+func (AppModule) ConsensusVersion() uint64 { return 2 }
+
+// BeginBlock returns the begin blocker for the accesscontrol module.
+func (am AppModule) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {}
```

Comment:
don't we want to refactor begin / end block to instead be finalize block? (not sure what that pattern looks like with abci++ or if under the hood cosmos sdk still just uses begin and end block?

Reply:
i took a look at the other module.go files and they're all using begin/end block -- it seems like FinalzieBlock is called in abci.go in the base app so it's abstracted

Reply:
Seems like under the hood tendermint makes an RPC call to the app:
 
https://github.com/sei-protocol/sei-cosmos/blob/main/baseapp/abci.go#L955 

Finalize block has a bunch of RPC references 
https://github.com/sei-protocol/sei-tendermint/search?q=FinalizeBlock

Whereas BeginBlock/EndBlock is mainly just markdown or testing
https://github.com/sei-protocol/sei-tendermint/search?q=FinalizeBlock